runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/JettyServletEngineAdapter.java

```diff
@@ -118,27 +120,45 @@ public InvocationType getInvocationType() {
             return InvocationType.BLOCKING;
           }
         };
-    rpcConnector =
-        new DelegateConnector(server, "RPC") {
-          @Override
-          public void run(Runnable runnable) {
-            // Override this so that it does the initial run in the same thread.
-            // Currently, we block until completion in serviceRequest() so no point starting new
-            // thread.
-            runnable.run();
-          }
-        };
-    server.addConnector(rpcConnector);
+
+    // Don't add the RPC Connector if in HttpConnector mode.
+    if (!isHttpConnectorMode)
+    {
+      rpcConnector =
+          new DelegateConnector(server, "RPC") {
+            @Override
+            public void run(Runnable runnable) {
+              // Override this so that it does the initial run in the same thread.
+              // Currently, we block until completion in serviceRequest() so no point starting new
+              // thread.
+              runnable.run();
+            }
+          };
+
+      HttpConfiguration httpConfiguration = rpcConnector.getHttpConfiguration();
+      httpConfiguration.setSendDateHeader(false);
+      httpConfiguration.setSendServerVersion(false);
+      httpConfiguration.setSendXPoweredBy(false);
+      if (LEGACY_MODE) {
+        httpConfiguration.setRequestCookieCompliance(CookieCompliance.RFC2965);
+        httpConfiguration.setResponseCookieCompliance(CookieCompliance.RFC2965);
+        httpConfiguration.setUriCompliance(UriCompliance.LEGACY);
+      }
+
```

Comment:
> What are your thoughts about movign all this to the constructor of DelegateConnector itself?

I wrote the jetty-delegate code to be a generic mechansim not google specific things, which could potentially be applied to other protocols other than the google RPC, so that it could potentially be extracted out into Jetty to be maintained in the future.

The interface with google code is done in the `com.google.apphosting.runtime.jetty.delegate.impl` package.

So I think adding compliance modes specific for the google usage should be here rather than in the `DelegateConnector`.